jupyterlab_gcedetails/src/components/hardware_configuration_dialog.tsx

```diff
@@ -127,8 +159,19 @@ export class HardwareConfigurationDialog extends React.Component<Props, State> {
             notebookService={notebookService}
             onCompletion={onCompletion}
             detailsServer={detailsServer}
+            onError={(err, errorType, instanceDetails?) =>
```

Comment:
I'm a bit confused why there are two callbacks passed in that both seem to pertain to error handling. Could the `onError` callback serve both purposes?

Reply:
The onError callback stores the error that was experienced in the state of the hardware scaling dialog. The showPage callback changes the view of the hardware scaling dialog to be the error page rather than the scaling status page. I have these separated because specifically if an error occurs while setting the machineType/accelerator we want to save the error in the state variable so we can pass it to the error page later but we don't want to show the error page until the start operation has been attempted. I thought it was more modular this way but I can also wrap these two functionalities into one callback that checks what type of error was received and doesn't change the hardware scaling dialog view if the error occurred while setting the machineType/accelerator to make it more concise if that's the better approach?

Reply:
I think this goes along with other comment about the HardwareScalingStatus component being responsible for both the operation and its display. The strange part for me is that the child component is bubbling up a status with the `onError` callback, but also being responsible for knowing when to change its parents state by calling `showErrorPage`. It seems like the parent component should be responsible for knowing when to transition states like it does for the other dialog states.

Reply:
Maybe it would fair better to move some of the logic in the scaling status component into the dialog component so the scaling status page is just updating the view based on the status of the dialog? We can talk more about this in our meeting